python/pyspark/pandas/indexing.py

```diff
@@ -1311,7 +1312,9 @@ def _select_cols_by_iterable(
                     % (len(cast(Sized, cols_sel)), len(self._internal.column_labels))
                 )
             if isinstance(cols_sel, pd.Series):
-                if not cols_sel.index.sort_values().equals(self._psdf.columns.sort_values()):
+                if get_option("compute.eager_check") and not cols_sel.index.sort_values().equals(
+                    self._psdf.columns.sort_values()
```

Comment:
Actually, I think this is fine @Yikun.  `self._psdf.columns` returns a pandas index. So sort_values on the column labels should be not so expensive in general.

Reply:
Ah, yes, and there are some differet behaviour on pandas and pandas on spark.

I hit this problem because pdf[[True, Fasle, True]], in pandas is rowselect, but for pandas on spark [is colselect](https://github.com/apache/spark/blob/master/python/pyspark/pandas/indexing.py#L805-L806). So colselect it's okay in here maybe.

Should I do a next commits to fix above comments or maybe just close this.

Reply:
Oh I see .. we should probably fix that case then.. maybe let's create a new PR to address that problem first, and then get this one merged after that.

Reply:
cc @ueshin FYI